doc/topics/async-system.md

```diff
@@ -0,0 +1,209 @@
+
+In practice, though, we almost never call `waitInMainThread`. After all, what good is it to start an asynchronous operation if we're just going to block the calling thread waiting for it? Instead, we register a _continuation function_ with the `Future`.
+
+## Continuation Functions {#continuation-functions}
+
+A continuation function is a callback - usually a lambda - that is invoked when the `Future` resolves. The name comes from the idea that the execution "continues" with that function after the async operation. Continuations are registered with the `then...` methods on the `Future`. Different `then` functions are used to control _in what thread_ the callback is invoked:
+
+* [thenInMainThread](@ref CesiumAsync::Future::thenInMainThread): The continuation function is invoked in the "main" thread. Many applications have a clear idea of which thread they consider the main one, but in Cesium Native, "main thread" very specifically means the thread that called `waitInMainThread` or [dispatchMainThreadTasks](@ref CesiumAsync::AsyncSystem::dispatchMainThreadTasks) on the `AsyncSystem`. This need not be the same as the thread your application considers to be the main one.
+* [thenInWorkerThread](@ref CesiumAsync::Future::thenInWorkerThread): The continuation function is invoked in a background worker thread by calling [startTask](@ref CesiumAsync::ITaskProcessor::startTask) on the `ITaskProcessor` instance with which the `AsyncSystem` was constructed.
+* [thenInThreadPool](@ref CesiumAsync::Future::thenInThreadPool): The continuation function is invoked in a specified [ThreadPool](@ref CesiumAsync::ThreadPool).
+* [thenImmediately](@ref CesiumAsync::Future::thenImmediately): The continuation function is invoked immediately in whatever thread happened to resolve the `Future`. This is only appropriate for work that completes very quickly and is safe to run in any thread.
+
+Continuing with our web request example above, we can register a continuation that receives the completed request and does something with the downloaded data in the main thread:
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp continuation
+
+So far, this all looks like a glorified callback system. The real power of `AsyncSystem` becomes apparent when we chain continuations together. Here's what that might look like:
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp chaining
+
+In this example, we call `get` to start a network request, as before. We immediately call `thenInWorkerThread` on the returned `Future` in order to register a continuation for when the request completes. This continuation runs in a background thread and calls a made-up function for this example, `processDownloadedContent`. We can imagine that it's doing some intensive processing, and we wouldn't want to block the main thread while it's doing it.
```

Comment:
I don't think you need to call out functions as "made-up" as all the code is a big example.

Reply:
I thought it was important to distinguish real parts of the Cesium Native API from functions made up for the purpose of the example. I didn't have this initially, but upon re-reading it with fresh eyes I thought it wasn't very obvious, so I tried to make it more so. Do you feel that referring to this function as "made up" detracts from the readability significantly? If not, I'm inclined to leave it.

Reply:
The real functions have Doxygen links, or they should, right? Even if not, it seems quite clear that `processDownloadedContent` comes from the example, and I find tagging it as "made-up" distracting.

Reply:
The pattern I've been following - inherited from standard practice at AGI 15 years ago! - is to make the first mention of a type a link, and successive mentions just `monospaced`. So the lack of a link doesn't clearly identify fake types and functions. As long as I'm in an AGI state of mind, the approach we used to use for made-up functions in documentation, as suggested by the late, great Ralph Patrick, was to name them things like `startAnyOldOperationThatMightFail` or `performAnyOldProcessingOnDownloadedContent`.
https://help.agi.com/STKComponents/html/AccessQueries.htm

I guess you're right that the potential for confusion isn't too high. So I'll remove "made-up" bits. The first time a user asks why they can't find the `startOperationThatMightFail` or `findReferenceImageUrl` function, though, I'm blaming you. 😜

Reply:
Yes, you can send them my email 😎